src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -118,6 +121,24 @@ internal void Freeze()
 
         private bool IsFrozen => _frozen != 0;
 
+        /// <summary>
+        /// This is the language-specific type symbol representing the private
+        /// implementation details class. Setter uses Interlocked.CompareExchange so that the
+        /// language proxy is set atomically, therefore the client should always
+        /// use the value from the getter after setting.
+        /// </summary>
```

Comment:
The semantics for this property are kind of weird as a whole -- it feels like what we want is a builder-like pattern, where it can be manipulated at will until it is frozen.